hugolib/pagination.go

```diff
@@ -527,6 +527,11 @@ func newPaginationURLFactory(d targetPathDescriptor) paginationURLFactory {
 
 		targetPath := createTargetPath(pathDescriptor)
 		targetPath = strings.TrimSuffix(targetPath, d.Type.BaseFilename())
+
+		if d.PathSpec.RemoveTrailingSlashesFromURLs {
+			targetPath = strings.TrimSuffix(targetPath, "/")
+		}
```

Comment:
What wuld be better would be to put this config option into the `pathDescriptor`. This way you can handle it in one place and add test(s) here:

https://github.com/spf13/hugo/blob/master/hugolib/page_paths_test.go#L40